File: src/main/java/io/tatum/model/response/offchain/WithdrawalResponse.java
package io.tatum.model.response.offchain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
@EqualsAndHashCode
public class WithdrawalResponse {

    /**
     * Transaction reference of the transaction connected to this withdrawal.
     *
     * @type {string}
     * @memberof WithdrawalResponse
     */
    private String reference;

    /**
     * @type {Array<WithdrawalResponseData>}
     * @memberof WithdrawalResponse
     */
    private WithdrawalResponseData[] data;
    /**
     * ID of withdrawal
     *
     * @type {string}
     * @memberof WithdrawalResponse
     */
    private String id;

}

File: src/main/kotlin/model/response/offchain/WithdrawalResponse.kt
package model.response.offchain

import java.math.BigDecimal
import java.math.BigInteger

/**
 * @property id ID of withdrawal
 * @property data
 * @property reference Transaction reference of the transaction connected to this withdrawal.
 */
class WithdrawalResponse(
    val reference: String,
    val data: Array<WithdrawalResponseData>,
    val id: String,
)

/**
 *
 * @property address
 * @property amount Amount of unprocessed transaction outputs, that can be used for withdrawal. Bitcoin, Litecoin, Bitcoin Cash only.
 * @property vIn Last used unprocessed transaction output, that can be used.
 * @property vInIndex Index of last used unprocessed transaction output in raw transaction, that can be used. Bitcoin, Litecoin, Bitcoin Cash only.
 * @property scriptPubKey Script of last unprocessed UTXO. Bitcoin SV only.
 */
class WithdrawalResponseData(
    val address: Address,
    val amount: BigDecimal,
    val vIn: String,
    val vInIndex: BigInteger,
    val scriptPubKey: String,
)
